OrbitClientServices/ProcessManager.cpp

```diff
@@ -85,7 +89,7 @@ void ProcessManagerImpl::Start() {
   worker_thread_ = std::thread([this] { WorkerFunction(); });
 }
 
-void ProcessManagerImpl::Shutdown() {
+void ProcessManagerImpl::ShutdownAndWait() noexcept {
   shutdown_mutex_.Lock();
   shutdown_initiated_ = true;
   shutdown_mutex_.Unlock();
```

Comment:
ok, this is not safe - worker_thread_.join() below can throw an exception. Maybe we need JoinThreadOrDie function here which does not throw.

Reply:
I think it's fine to just swallow the exception in that case. Something like:
```
try {
worker_thread_.join();
catch (const std::exception&) {}
```

@dimitry- WDYT?

Reply:
well, I think we kind of want to add FATAL in exception handling part at least, preferably checking for conditions before too.

Reply:
I did not add another JoinThreadOrDie function, but catch the potential exception there. ptal